Mathlib/AlgebraicGeometry/Modules/Tilde.lean

```diff
@@ -158,7 +158,10 @@ noncomputable instance (U : (Opens (PrimeSpectrum.Top R))ᵒᵖ) :
 noncomputable def tilde : (Spec (CommRingCat.of R)).Modules where
   val :=
     { obj := fun U ↦ ModuleCat.of _ (M.tildeInType.val.obj U)
-      map := fun {U V} i ↦
+      map := fun {U V} i ↦ asHom
+        (X := of ((Spec (CommRingCat.of R)).ringCatSheaf.val.obj U) (M.tildeInType.val.obj U))
+        (Y := (restrictScalars ((Spec (CommRingCat.of R)).ringCatSheaf.val.map i)).obj
+          (of ((Spec (CommRingCat.of R)).ringCatSheaf.val.obj V) (M.tildeInType.val.obj V)))
```

Comment:
Having to write out the source and target so often doesn't seem fun. :-(

At very least we should leave some comments on the places this is necessary, reminding us to come back.

I wonder if having a variant of `asHom` here, that took as input a semilinear map (twisted by `(Spec (CommRingCat.of R)).ringCatSheaf.val.map i`), could producing something that would typecheck more easily... Not sure yet.

Reply:
Something like
```
def tildeInTypeLinearMap {U V : (Opens (PrimeSpectrum.Top R))ᵒᵖ} (i : U ⟶ V) :
    (M.tildeInType.1.obj U) →ₛₗ[(Spec (CommRingCat.of R)).ringCatSheaf.val.map i] 
        (of ((Spec (.of R)).ringCatSheaf.1.obj V) (M.tildeInType.1.obj V)) :=
  { toFun := M.tildeInType.val.map i
    map_smul' := by intros; rfl
    map_add' := by intros; rfl } 

def asHomRestrictScalars {M N : Type u} {R S : Type*} [Ring R] [Ring S] (σ : R →+* S)
    [AddCommGroup M] [Module R M] [AddCommGroup N] [Module S N] 
    (f : M →ₛₗ[σ] N) :
    ModuleCat.of R M ⟶ (restrictScalars σ).obj (ModuleCat.of S N) :=
  sorry

/--
`M^~` as a sheaf of `𝒪_{Spec R}`-modules
-/
noncomputable def tilde : (Spec (CommRingCat.of R)).Modules where
  val :=
    { obj := fun U ↦ ModuleCat.of _ (M.tildeInType.val.obj U)
      map := fun {U V} i ↦ 
        asHomRestrictScalars ((Spec (CommRingCat.of R)).ringCatSheaf.val.map i) (tildeInTypeLinearMap M i)
      map_id := sorry
      map_comp := sorry }
  isSheaf := (TopCat.Presheaf.isSheaf_iff_isSheaf_comp (forget AddCommGrp) _ ).2
    M.tildeInType.2
```

Reply:
Why is `restrictScalars` printing with a `coe` arrow?? That is really unhelpful. :-)

Reply:
I went through and added a comment at each point we have to add a type hint: looks like `restrictScalars` is always involved somehow.